crates/bevy_reflect/src/serde/de.rs

```diff
@@ -29,260 +47,273 @@ impl<'a, 'de> DeserializeSeed<'de> for ReflectDeserializer<'a> {
     where
         D: serde::Deserializer<'de>,
     {
-        deserializer.deserialize_any(ReflectVisitor {
+        deserializer.deserialize_map(ReflectDeserializerVisitor {
             registry: self.registry,
         })
     }
 }
 
-struct ReflectVisitor<'a> {
+struct ReflectDeserializerVisitor<'a> {
     registry: &'a TypeRegistry,
 }
 
-impl<'a, 'de> Visitor<'de> for ReflectVisitor<'a> {
+impl<'a, 'de> Visitor<'de> for ReflectDeserializerVisitor<'a> {
```

Comment:
This probably wouldn't work since we need the type name in order to deserialize the actual value (plus this makes it use struct format which might not be desired)